### ISSUE TITLE
Issues while training on other dataset

### ISSUE THREAD

**Issue by bhagyashreerawl** [XifengGuo/CapsNet-Keras#75]
@XifengGuo ,

Thank you posting this implementation. I was trying to implement this network on distracted driver detection dataset. This dataset has also 10 classes. I changed the load mnist function and converted the images into grayscale images.  Ehile training, the loss output and the decoder_loss output gives value as nan. Can you please shed some light to why this must be happening?

Thank you in advance.

Regards,
Bhagyashree

**Comment by XifengGuo** [comment]
@bhagyashreerawl Is the image size in your dataset 28 by 28? Have you scaled them to [0,1]?

**Comment by bhagyashreerawl** [comment]
@XifengGuo 
Yes it is 28 by 28. By scaling them, you mean one hot encoding right? 
Yes they are one hot encoded.

Regards,
Bhagyashree

**Comment by bhagyashreerawl** [comment]
@XifengGuo 
I was converting them earlier, but there were some errors. Thank you so much. 
I have one more question. Can you please explain the difference between capsnet_loss, loss and decoder_loss and what does capsnet accuracy refers to?
I do have an idea about all this. Just want to get a clearer understanding.

Thank you once again for taking out time to give a response.

Regards,
Bhagyashree

**Comment by XifengGuo** [comment]
capsnet_loss is the classification loss, i.e., margin loss defined by Eq. (4) in the paper. decoder_loss is the reconstruction loss of autoencoder, and loss is the weighted sum of capsnet_loss and decoder_loss.
